django/contrib/redirects/forms.py

```diff
@@ -0,0 +1,42 @@
+from django import forms
+from django.apps import apps
+from django.contrib import admin
+from django.contrib.redirects.models import Redirect
+from django.utils.translation import gettext_lazy as _
+
+
+class RedirectForm(forms.ModelForm):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if apps.is_installed('django.contrib.sites'):
+            from django.contrib.sites.models import Site
+
+            self.fields['site'] = forms.ModelChoiceField(
+                queryset=Site.objects.all(),
+                to_field_name='domain',
+                empty_label='Any',
+                widget=admin.widgets.AdminRadioSelect(attrs={
+                    'class': admin.options.get_ul_class(admin.VERTICAL),
+                }),
+                label=_('Site'),
+                help_text=_('If domain is not set, redirect from this site.'),
+                required=False,
+            )
+
+    def save(self, commit=True):
+        instance = super().save(commit=False)
+
+        if apps.is_installed('django.contrib.sites') and not instance.domain:
```

Comment:
I would recommend putting this inside a `clean` function, not in save.

Reply:
There is no form validation, so why should this be in clean?

Reply:
`clean` is not only for validation, but also for data modification in a form.